lib/modules/manager/osgi/extract.ts

```diff
@@ -0,0 +1,135 @@
+import is from '@sindresorhus/is';
+import * as json5 from 'json5';
+import { coerce, satisfies } from 'semver';
+import { logger } from '../../../logger';
+import { MavenDatasource } from '../../datasource/maven';
+import type { ExtractConfig, PackageDependency, PackageFile } from '../types';
+import type { Bundle, FeatureModel } from './types';
+
+export function extractPackageFile(
+  content: string,
+  fileName: string,
+  config?: ExtractConfig
+): PackageFile | null {
+  // References:
+  // - OSGi compendium release 8 ( https://docs.osgi.org/specification/osgi.cmpn/8.0.0/service.feature.html )
+  // - The Sling implementation of the feature model ( https://sling.apache.org/documentation/development/feature-model.html )
+  logger.debug({ fileName }, 'osgi.extractPackageFile');
+
+  const deps: PackageDependency[] = [];
+  let featureModel: FeatureModel;
+  try {
+    // Compendium R8 159.3: JS comments are supported
+    featureModel = json5.parse<FeatureModel>(content);
+  } catch (e) {
+    logger.warn(`Failed parsing ${fileName}: ${(e as Error).message}`);
```

Comment:
```suggestion
  } catch (err) {
    logger.warn({ fileName, err }, 'Failed to parse osgi file');
```

For warnings and errors, we want the `message` to be static so that it can be aggregated or ignored in people's log systems.

If this is likely to happen often and be ignorable then it could just be a debug message instead of warn.